test/extended/operators/olm.go

```diff
@@ -251,50 +251,30 @@ var _ = g.Describe("[Feature:Platform] an end user use OLM", func() {
 		}
 	})
 
-	// OCP-27589 do not use ipv4 addresses in CatalogSources generated by marketplace
 	// author: tbuskey@redhat.com
-	g.It("do not use ipv4 addresses in CatalogSources generated by marketplace", func() {
-		re := regexp.MustCompile(`(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}`)
-		olmErrs := 0
+	g.It("OLM-Low-OCP-24058-components should have resource limits defined", func() {
+		olmUnlimited := 0
 		olmNames := []string{""}
-		olmNamespace := "openshift-marketplace"
-		olmJpath := "-o=jsonpath={range .items[*]}{@.metadata.name}{','}{@.spec.address}{'\\n'}"
-		msg, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("catalogsource", "-n", olmNamespace, olmJpath).Output()
+		olmNamespace := "openshift-operator-lifecycle-manager"
+		olmJpath := "-o=jsonpath={range .items[*]}{@.metadata.name}{','}{@.spec.containers[0].resources.requests.*}{'\\n'}"
+		msg, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("pod", "-n", olmNamespace, olmJpath).Output()
 		if err != nil {
 			e2e.Failf("Unable to get pod -n %v %v.", olmNamespace, olmJpath)
 		}
 		o.Expect(err).NotTo(o.HaveOccurred())
 		o.Expect(msg).NotTo(o.ContainSubstring("No resources found"))
-		// msg = fmt.Sprintf("%v\ntest,1.1.1.1\n", msg)
 		lines := strings.Split(msg, "\n")
 		for _, line := range lines {
-			if len(line) <= 0 {
-				continue
-			}
 			name := strings.Split(line, ",")
-			cscAddr := strings.Split(name[1], ":")[0]
-			if re.MatchString(cscAddr) {
-				olmErrs++
-				olmNames = append(olmNames, name[0])
+			if len(name) > 1 {
+				if len(name) > 1 && len(name[1]) < 1 {
+					olmUnlimited++
+					olmNames = append(olmNames, name[0])
+				}
 			}
 		}
-		if olmErrs > 0 {
-			e2e.Failf("%v ipv4 addresses found in these OLM components: %v", olmErrs, olmNames)
-		}
-	})
-
-	// OCP-21130 - [bug ALM-736] Fetching non-existent `PackageManifest` should return 404
-	// author: bandrade@redhat.com
-	g.It("Fetching non-existent `PackageManifest` should return 404", func() {
-		msg, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("packagemanifest", "--all-namespaces", "--no-headers").Output()
-		o.Expect(err).NotTo(o.HaveOccurred())
-		packageserverLines := strings.Split(msg, "\n")
-		if len(packageserverLines) > 0 {
-			raw, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("packagemanifest", "a_package_that_not_exists", "-o yaml", "--loglevel=8").Output()
-			o.Expect(err).To(o.HaveOccurred())
-			o.Expect(raw).To(o.ContainSubstring("\"code\": 404"))
-		} else {
-			e2e.Failf("No packages to evaluate if 404 works when a PackageManifest does not exists")
+		if olmUnlimited > 0 {
+			e2e.Failf("There are no limits set on %v of %v OLM components: %v", olmUnlimited, len(lines), olmNames)
```

Comment:
Because we check the `request` resource, it's better to replace the `limits` with `requests` string.